staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -458,6 +458,18 @@ func (c *AvailableConditionController) updateAPIServiceStatus(originalAPIService
 		return newAPIService, nil
 	}
 
+	orig := apiregistrationv1apihelper.GetAPIServiceConditionByType(originalAPIService, apiregistrationv1.Available)
+	now := apiregistrationv1apihelper.GetAPIServiceConditionByType(newAPIService, apiregistrationv1.Available)
+	if orig == nil {
+		orig = &apiregistrationv1.APIServiceCondition{
+			Type:   apiregistrationv1.Available,
+			Status: apiregistrationv1.ConditionUnknown,
+		}
+	}
+	if *orig != *now {
```

Comment:
What if now is nil? (Eg. the test failure)
goroutine 148 [running]:
testing.tRunner.func1.2(0x1f2e340, 0x2e1d980)
	/usr/local/go/src/testing/testing.go:1143 +0x49f
testing.tRunner.func1(0xc0004b8300)
	/usr/local/go/src/testing/testing.go:1146 +0x695
panic(0x1f2e340, 0x2e1d980)
	/usr/local/go/src/runtime/panic.go:971 +0x499
k8s.io/kubernetes/vendor/k8s.io/kube-aggregator/pkg/controllers/status.(*AvailableConditionController).updateAPIServiceStatus(0xc0000c9e00, 0xc0004b8480, 0xc0004b8600, 0x0, 0x0, 0x0)
	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go:469 +0x1f1
k8s.io/kubernetes/vendor/k8s.io/kube-aggregator/pkg/controllers/status.TestUpdateAPIServiceStatus(0xc0004b8300)
	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/kube-aggregator/pkg/controllers/status/available_controller_test.go:473 +0x385
testing.tRunner(0xc0004b8300, 0x2186808)
	/usr/local/go/src/testing/testing.go:1193 +0x203
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1238 +0x5d8

Reply:
fixed